### ISSUE TITLE
matchplay false postivies cannot be disabled with # noqa

### ISSUE THREAD

**Issue by cdesouza71** [ansible/ansible-lint#1911]
##### Summary

False positives reported by matchplay cannot be disabled with # noqa rule-name.

##### Issue Type

- Bug Report

##### Ansible and Ansible Lint details

```console
qc@qc-pde:$ ansible --version
ansible [core 2.11.5] 
  config file = None
  configured module search path = ['/home/qc/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/lib/python3.8/dist-packages/ansible
  ansible collection location = /home/qc/.ansible/collections:/usr/share/ansible/collections
  executable location = /usr/local/bin/ansible
  python version = 3.8.10 (default, Nov 26 2021, 20:14:08) [GCC 9.3.0]
  jinja version = 3.0.1
  libyaml = True

qc@qc-pde:$ ansible-lint --version
WARNING: PATH altered to include /usr/bin
ansible-lint 5.4.0 using ansible 2.11.5
```

- ansible installation method: pip
- ansible-lint installation method: pip

##### OS / ENVIRONMENT

```console
Distributor ID:	Ubuntu
Description:	Ubuntu 20.04.3 LTS
Release:	20.04
Codename:	focal
```

##### STEPS TO REPRODUCE

```console 
qc@qc-pde:$ ansible-lint -vvv -p -r rules/  data/task-in-playbook.yml
```

rules/TaskInPlaybookRule.py
```python 
from ansiblelint.rules import AnsibleLintRule

class TaskInPlaybookRule(AnsibleLintRule):
    id = "task-in-playbook"
    shortdesc = "Tasks should go in roles not the main playbook"
    description = (
        "Tasks should normally go into roles rather than our top level "
        + "playbooks. This is to keep the playbooks simple and make the code reusable. "
    )

    def matchplay(self, playbookfile, play):
        if "tasks" in play:
            matches = []
            for t in play["tasks"]:
                match = "Task '{}' in playbook".format(t.get("name", ""))
                matches.append(
                    self.create_matcherror(
                        message=match, filename=playbookfile, linenumber=t["__line__"]
                    )
                )
            return matches
        return []
```

data/task-in-playbook.yml
```yml
---
- name: Playbooks contains task not masked
  hosts: localhost
  tasks:
    - name: Check if directory exists
      stat:
        path: "/tmp"

- name: Playbooks contains task masked
  hosts: localhost
  tasks:
    - name: Check if directory exists  # noqa task-in-playbook
      stat:
        path: "/tmp"
```

##### Desired Behavior

The first task in the playbook should fail linting whereas the second task should pass linting as it is masked with `# noqa`.

##### Actual Behavior

Both tasks in the playbook fail linting.

```console
WARNING: PATH altered to include /usr/bin
DEBUG    Logging initialized to level 10
DEBUG    Options: Namespace(cache_dir='/home/qc/.cache/ansible-lint/dc1163', colored=True, config_file='/home/qc/projects/linter-bug/dev/docker/ansible-linter/custom-rules/.ansible-lint', configured=True, cwd=PosixPath('/home/qc/projects/linter-bug/dev/docker/ansible-linter/custom-rules/bug'), display_relative_path=True, enable_list=[], exclude_paths=['/home/qc/projects/linter-bug/dev/docker/ansible-linter/custom-rules/dummy'], extra_vars=None, format='rich', kinds=[{'jinja2': '**/*.j2'}, {'jinja2': '**/*.j2.*'}, {'inventory': '**/inventory/**.yml'}, {'requirements': '**/meta/requirements.yml'}, {'galaxy': '**/galaxy.yml'}, {'reno': '**/releasenotes/*/*.{yaml,yml}'}, {'playbook': '**/playbooks/*.{yml,yaml}'}, {'playbook': '**/*playbook*.{yml,yaml}'}, {'role': '**/roles/*/'}, {'tasks': '**/tasks/**/*.{yaml,yml}'}, {'handlers': '**/handlers/*.{yaml,yml}'}, {'vars': '**/{host_vars,group_vars,vars,defaults}/**/*.{yaml,yml}'}, {'meta': '**/meta/main.{yaml,yml}'}, {'yaml': '.config/molecule/config.{yaml,yml}'}, {'requirements': '**/molecule/*/{collections,requirements}.{yaml,yml}'}, {'yaml': '**/molecule/*/{base,molecule}.{yaml,yml}'}, {'requirements': '**/requirements.yml'}, {'playbook': '**/molecule/*/*.{yaml,yml}'}, {'yaml': '**/{.ansible-lint,.yamllint}'}, {'yaml': '**/*.{yaml,yml}'}, {'yaml': '**/.*.{yaml,yml}'}], lintables=['data/task-in-playbook.yml'], listrules=False, listtags=False, loop_var_prefix=None, mock_modules=[], mock_roles=[], offline=False, parseable=True, parseable_severity=False, progressive=False, project_dir='..', quiet=False, rules={}, rulesdir=[PosixPath('/home/qc/projects/linter-bug/dev/docker/ansible-linter/custom-rules/bug/rules')], rulesdirs=[PosixPath('/home/qc/projects/linter-bug/dev/docker/ansible-linter/custom-rules/bug/rules')], skip_action_validation=True, skip_list=[], tags=[], use_default_rules=False, var_naming_pattern=None, verbosity=3, version=False, warn_list=['experimental', 'role-name'])
DEBUG    /home/qc/projects/linter-bug/dev/docker/ansible-linter/custom-rules/bug
DEBUG    Loading rules from /home/qc/projects/linter-bug/dev/docker/ansible-linter/custom-rules/bug/rules
INFO     Executing syntax check on data/task-in-playbook.yml (4.22s)
DEBUG    Examining data/task-in-playbook.yml of type playbook
WARNING  Listing 2 violation(s) that are fatal
data/task-in-playbook.yml:5: task-in-playbook Task 'Check if directory exists' in playbook
data/task-in-playbook.yml:12: task-in-playbook Task 'Check if directory exists' in playbook
You can skip specific rules or tags by adding them to your configuration file:
# .ansible-lint
warn_list:  # or 'skip_list' to silence them completely
  - task-in-playbook  # Tasks should go in roles not the main playbook

Finished with 2 failure(s), 0 warning(s) on 1 files.
```
